src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyFlowNode.cs

```diff
@@ -64,11 +66,22 @@ public void CalculateLongestPathTime()
                     edgesOfInterest = OutgoingEdges;
                 }
 
-                double worstCasePathTime = edgesOfInterest.Max(e => e.To.WorstCasePathTime + e.To.PrBuildTime) + OfficialBuildTime;
-                WorstCasePathTime = worstCasePathTime > WorstCasePathTime ? worstCasePathTime : WorstCasePathTime;
+                // Tooling subscriptions should not be included in longest path calculation
+                edgesOfInterest = edgesOfInterest.Where(e => !e.IsToolingOnly).ToList();
```

Comment:
You can just add this where clause to line 61

```
var edgesOfInterest = OutgoingEdges.Where(e => !e.BackEdge && !e.IsToolingOnly).ToList();
```

Reply:
I think we can't because there is an `if` condition on line 64 that will work incorrectly if we modify that query:
```
if (edgesOfInterest.Count == 0)
{
    edgesOfInterest = OutgoingEdges;
}
```